kayak_core/src/widget_manager.rs

```diff
@@ -406,6 +471,45 @@ impl WidgetManager {
         self.nodes[*id].clone()
     }
 
+    /// Bind a widget so that it re-renders when the binding changes
+    ///
+    /// # Arguments
+    ///
+    /// * `id`: The ID of the widget
+    /// * `binding`: the binding to watch
+    ///
+    pub(crate) fn bind<T>(&mut self, id: Index, binding: &Binding<T>)
```

Comment:
In a future PR, we'll probably have `KayakContext` use this method for state management rather than having its own duplicate logic for that.